File: php-sample/ad_management_sample.php
<?php
ini_set("default_socket_timeout", 30);
require_once 'restapi.php';

$config = parse_ini_file("sample.ini");
print_r($config);

function debug($obj, $detail = false)
{
    if (is_array($obj)) {
        echo "size : " . count($obj) . "\n";
    }
    if ($detail) {
        print_r($obj);
    }
}

// #. detail log
$DEBUG = false;

$api = new RestApi($config['BASE_URL'], $config['API_KEY'], $config['SECRET_KEY'], $config['CUSTOMER_ID']);

echo "Test ManagedCustomerLink\n";
$customerlist = $api->GET('/customer-links', array('type' => 'MYCLIENTS'));
debug($customerlist, $DEBUG);

echo "Test BusinessChannel\n";
$businesschannellist = $api->GET('/ncc/channels');
debug($businesschannellist, $DEBUG);

echo "Test Campaign\n";
$campaignlist = $api->GET('/ncc/campaigns');
debug($campaignlist, $DEBUG);

echo "Test Adgroup\n";
echo "  #1. GET Adgroup\n";
$adgrouplist = $api->GET('/ncc/adgroups');
debug($adgrouplist, $DEBUG);
$targetadgroup = $adgrouplist[0];

echo "  #2. CREATE Adgroup\n";
$data = array(
    "name" => 'TEST_#' . rand(),
    "nccCampaignId" => $targetadgroup ["nccCampaignId"],
    "pcChannelId" => $targetadgroup ["pcChannelId"],
    "mobileChannelId" => $targetadgroup ["mobileChannelId"]
);
$createadgroup = $api->POST('/ncc/adgroups', $data);
debug($createadgroup, $DEBUG);

echo "  #3. UPDATE Adgroup\n";
$createadgroup["userLock"] = 1;
$updateadgroup = $api->PUT('/ncc/adgroups/' . $createadgroup["nccAdgroupId"], $createadgroup, array("fields" => 'userLock'));
debug($updateadgroup, $DEBUG);

echo "Test AdKeyword\n";
echo "  #1. CREATE AdKeyword\n";
$data = array(
    array(
        "keyword" => "hello2"
    )
);
$createkeyword = $api->POST('/ncc/keywords', $data, array("nccAdgroupId" => $updateadgroup["nccAdgroupId"]));
debug($createkeyword, $DEBUG);

echo "  #2. GET AdKeyword\n";
$adkeywordlist = $api->GET('/ncc/keywords', array("nccAdgroupId" => $updateadgroup["nccAdgroupId"]));
debug($adkeywordlist, $DEBUG);

echo "  #3. UPDATE AdKeyword\n";
$adkeywordlist[0]["userLock"] = 1;
$updatekeyword = $api->PUT('/ncc/keywords', $adkeywordlist, array("fields" => "userLock"));
debug($updatekeyword, $DEBUG);

echo "  #4. DELETE AdKeyword\n";
$api->DELETE('/ncc/keywords/' . $createkeyword[0]["nccKeywordId"]);

echo "  #4. DELETE Adgroup\n";
$api->DELETE('/ncc/adgroups/' . $createadgroup["nccAdgroupId"]);

echo "Test Estimate\n";
echo "  #1. average-position-bid\n";
$req_avg_pos = array(
    "device" => "PC",
    "items" => array(
        array("key" => "제주여행", "position" => 1),
        array("key" => "게스트하우스", "position" => 2),
        array("key" => "자전거여행", "position" => 3),
    )
);
$response = $api->POST('/estimate/average-position-bid/keyword', $req_avg_pos);
debug($response, $DEBUG);

echo "  #2. exposure-minimum-bid\n";
$req_bid = array(
    "device" => "PC",
    "period" => "MONTH",
    "items" => array(
        "제주여행",
        "게스트하우스",
        "자전거여행",
    )
);
$response = $api->POST('/estimate/exposure-minimum-bid/keyword', $req_bid);
debug($response, $DEBUG);

echo "  #3. median-bid\n";
$response = $api->POST('/estimate/median-bid/keyword', $req_bid);
debug($response, $DEBUG);

echo "  #4. performance\n";
$req_performance = array(
    "device" => "PC",
    "keywordplus" => true,
    "key" => "중고차",
    "bids" => array(
        100,
        500,
        1000,
        1500,
        2000,
        3000,
        5000,
    )
);
$response = $api->POST('/estimate/performance/keyword', $req_performance);
debug($response, $DEBUG);

echo "  #5. performance-bulk\n";
$req_performance_bulk = array (
		"items" => array (
				0 => array (
						"device" => "PC",
						"keywordplus" => true,
						"keyword" => "제주여행",
						"bid" => 70 
				),
				1 => array (
						"device" => "PC",
						"keywordplus" => true,
						"keyword" => "제주도",
						"bid" => 80 
				),
				2 => array (
						"device" => "PC",
						"keywordplus" => true,
						"keyword" => "제주맛집",
						"bid" => 90 
				) 
		) 
);
$response = $api->POST('/estimate/performance-bulk', $req_performance_bulk);
debug($response, $DEBUG);

echo "\nTest End\n";
?>

File: python-sample/examples/ad_management_sample.py
import time
import random
import requests

import signaturehelper


def get_header(method, uri, api_key, secret_key, customer_id):
    timestamp = str(round(time.time() * 1000))
    signature = signaturehelper.Signature.generate(timestamp, method, uri, SECRET_KEY)
    return {'Content-Type': 'application/json; charset=UTF-8', 'X-Timestamp': timestamp, 'X-API-KEY': API_KEY, 'X-Customer': str(CUSTOMER_ID), 'X-Signature': signature}


BASE_URL = 'https://api.naver.com'
API_KEY = '<API_KEY>'
SECRET_KEY = '<SECRET_KEY>'
CUSTOMER_ID = '<CUSTOMER_ID>'

# ManageCustomerLink Usage Sample

uri = '/customer-links'
method = 'GET'
r = requests.get(BASE_URL + uri, params={'type': 'MYCLIENTS'}, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))


# BusinessChannel Usage Sample

uri = '/ncc/channels'
method = 'GET'
r = requests.get(BASE_URL + uri, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))


# Adgroup Usage Sample

# 1. GET adgroup Usage Sample

uri = '/ncc/adgroups'
method = 'GET'
r = requests.get(BASE_URL + uri, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))
target_adgroup = r.json()[0]

# 2. CREATE adgroup Usage Sample

uri = '/ncc/adgroups'
method = 'POST'
payload = {'name': 'TEST#' + str(random.randrange(1000, 9999)), 'nccCampaignId' : target_adgroup['nccCampaignId'], 'pcChannelId' : target_adgroup['pcChannelId'], 'mobileChannelId': target_adgroup['mobileChannelId']}
r = requests.post(BASE_URL + uri, json=payload, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))

created_adgroup = r.json()

# 3. UPDATE Adgroup Usage Sample

uri = '/ncc/adgroups/' + created_adgroup['nccAdgroupId']
method = 'PUT'
created_adgroup['userLock'] = 0
r = requests.put(BASE_URL + uri, params={'fields': 'userLock'}, json=created_adgroup, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))

# 4. DELETE Adgroup

uri = '/ncc/adgroups/' + created_adgroup['nccAdgroupId']
method = 'DELETE'
r = requests.delete(BASE_URL + uri, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.content))

# AdKeyword Usage Sample

# 1. CREATE AdKeyword

uri = '/ncc/keywords'
method = 'POST'
r = requests.post(BASE_URL + uri, params={'nccAdgroupId': created_adgroup['nccAdgroupId']}, json=[{'keyword': 'hello'}], headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))

created_adkeyword = r.json()[0]

# 2. GET AdKeyword

uri = '/ncc/keywords'
method = 'GET'
r = requests.get(BASE_URL + uri, params={'nccAdgroupId': created_adgroup['nccAdgroupId']}, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))

# 3. UPDATE AdKeyword

uri = '/ncc/keywords'
method = 'PUT'
created_adkeyword['userLock'] = 0
r = requests.put(BASE_URL + uri, params={'fields': 'userLock'}, json=[created_adkeyword], headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))


# 4. GET and UPDATE AdKeyword (BidAmt)

uri = '/ncc/keywords/' + created_adkeyword['nccKeywordId']
method = 'GET'
r = requests.get(BASE_URL + uri, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))

retrieved_adkeyword = r.json()

uri = '/ncc/keywords'
method = 'PUT'
retrieved_adkeyword['bidAmt'] = 300
retrieved_adkeyword['useGroupBidAmt'] = 0
r = requests.put(BASE_URL + uri, params={'fields': 'bidAmt'}, json=[retrieved_adkeyword], headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))


# 5. DELETE AdKeyword

uri = '/ncc/keywords/' + created_adkeyword['nccKeywordId']
method = 'DELETE'
r = requests.delete(BASE_URL + uri, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.content))


# Estimate Usage Sample

# 1. average-position-bid

uri = '/estimate/average-position-bid/keyword'
method = 'POST'
r = requests.post(BASE_URL + uri, json={'device': 'PC', 'items': [{'key': '제주여행', 'position': 1}, {'key': '게스트하우스', 'position': 2}, {'key': '자전거여행', 'position': 3}]}, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("#response status_code = {}".format(r.status_code))
print("#response body = {}".format(r.json()))


# 2. exposure-minimum-bid

uri = '/estimate/exposure-minimum-bid/keyword'
method = 'POST'
r = requests.post(BASE_URL + uri, json={'device': 'PC', 'period': 'MONTH', 'items': ['제주여행', '게시트하우스', '자전거여행']}, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))

# 3. median-bid

uri = '/estimate/median-bid/keyword'
method = 'POST'
r = requests.post(BASE_URL + uri, json={'device': 'PC', 'period': 'MONTH', 'items': ['제주여행', '게시트하우스', '자전거여행']}, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))


# 4. performance

uri = '/estimate/performance/keyword'
method = 'POST'
r = requests.post(BASE_URL + uri, json={'device': 'PC', 'keywordplus': True, 'key': '중고차', 'bids': [100, 500, 1000, 1500, 2000, 3000, 5000]}, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))

# 5. performance-bulk

uri = '/estimate/performance-bulk'
method = 'POST'
r = requests.post(BASE_URL + uri, json={'items': [{'device': 'PC', 'keywordplus': True, 'keyword': '제주여행', 'bid': 70}, {'device': 'PC', 'keywordplus': True, 'keyword': '제주도', 'bid': 80}, {'device': 'PC', 'keywordplus': True, 'keyword': '제주도맛집', 'bid': 90}, ]}, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))

# Stat Usage Sample

# 1. GET Summary Report per multiple entities 

uri = '/stats'
method = 'GET'
stat_ids = [target_adgroup['nccCampaignId'], target_adgroup['nccAdgroupId']]
r = requests.get(BASE_URL + uri, params={'ids': stat_ids, 'fields': '["clkCnt","impCnt","salesAmt", "ctr", "cpc", "avgRnk", "ccnt"]', 'timeRange': '{"since":"2019-06-01","until":"2019-06-25"}'}, headers=get_header(method, uri, API_KEY, SECRET_KEY, CUSTOMER_ID))

print("response status_code = {}".format(r.status_code))
print("response body = {}".format(r.json()))
